akka-cluster-tools/src/main/scala/akka/cluster/pubsub/DistributedPubSubMediator.scala

```diff
@@ -258,6 +289,33 @@ object DistributedPubSubMediator {
 
     def encName(s: String) = URLEncoder.encode(s, "utf-8")
 
+    def mkKey(ref: ActorRef): String = mkKey(ref.path)
+
+    def mkKey(path: ActorPath): String = path.toStringWithoutAddress
+
+    trait PerGroupingBuffer {
+      var buffers: Map[String, Vector[(Any, ActorRef)]] = Map.empty
+
+      def totalBufferSize = buffers.foldLeft(0) { (sum, entity) ⇒ sum + entity._2.size }
```

Comment:
and then this can be done via a counter that is kept up-to-date, since folding creates lots of garbage and wastes cycles